proposals/0013-expanded-comparison-sampling.md

```diff
@@ -0,0 +1,441 @@
+ SPIRV intrinsics for pixel, compute, and node shader stages.
+Where the `DerivativesInMeshAndAmplificationShadersSupported` capability bit is
+ supported, verify that `SampleCmpBias` produces appropriate code including the
+ new DXIL and SPIRV intrinsics for mesh and amplification shader stages.
+For all of the above, verify that no validation errors are produced for the
+ generated code.
+For all of the above, wherever `SampleCmpGrad` and `SampleCmpBias` are used,
+ check that the appropriate shader flag is set.
+The above should all be doable with LIT filecheck tests.
+
+#### Diagnostics Testing
+
+Use `SampleCmpBias` in code reachable from the entry point of each unsupported
+ shader stage (vertex, geometry, tesselation stages, ray tracing stages, and
+ mesh stages if lacking hardware support for mesh derivatives)
+Using `SampleCmpBias`, `SampleCmpGrad`, and the new `CalculateLevelOfDetail*`
+ overloads in an unsupported shader model on a supported stage and verify that
+ a special error indicating the require version is produced.
+
+### Validation Testing
+
+Test that the Bias range limits tested by `SampleBiasFail` are respected by
+ `dx.op.sampleCmpBias`. (see Instr.ImmBiasForSampleB)
+Test that DXIL validation of allowed resource types must be performed for
+  `dx.op.sampleCmpBias` and `dx.op.sampleCmpGrad`:
+
+* Instr.SampleCompType - texture resources are declared with types that are
+  sampleable in the current environment.
+* Instr.SamplerModeForSampleC - sampler resource used by `SampleCmpBias` and
+  `SampleCmpGrad` are of type `SamplerComparisonState`.
+* Instr.ResourceKindForCalcLOD  - texture resources used by
+  CalculateLevelOfDetail* are among the valid texture objects.
+* Instr.ResourceKindForSample  - texture resources used by
+  `SampleCmpBias` and `SampleCmpGrad` are among the valid texture objects.
+* Instr.ResourceClassForSamplerGather - texture resources used by
+  `SampleCmpBias` and `SampleCmpGrad` must be SRVs.
+
+### Execution Testing
+
+Since there is no defined equation to determine which mip level an
+ implementation should use, the best approach is to compare the results of
+ different operations involving mip levels to ensure that they are consistent.
+
+To test `CalculateLevelOfDetail*` methods,
+use a texture with sentinel values in each mip level and ensure that the mip
+  level returned by CalculateLevelOfDetail matches the behavior of the original
+  SampleCmp operation by providing the sentinel value of the mip level
+  CalculateLevelOfDetail* made you expect and making sure that it matches.
+This will ensure that CalculateLevelOfDetail correctly returns the mip value
+that sample would have used.
+Ensure as well that `CalculateLevelOfDetail` clamps the mip result and
+ `CalculateLevelOfDetailUnclamped` does not.
+
+Given the nature of SampleCmp operations, the tests will have to be performed
+ with comparison values and success determined based on whether it matches what
+ is expected.
```

Comment:
Considering the parallel nature of shaders, couldn't this be performed without iteration in a single pass in a pixel shader?

For instance, you could output several values from each pixel in the pixel shader, then check that the values are within expected tolerances.  The values could consist of 1. a reference value (such as the result of CalculateLOD, no bias), 2. the comparison result with bias, 3. the bias value, which could be an input interpolated on one RT coordinate dimension (like X).  On the other RT coordinate dimension (like Y), you could vary something that impacts the scale of the input coordinate gradient to provide a range of derivative scales, and thus LODs, across that RT dimension.  Additionally, you could include a sampled result with non-comparison sampler as a reference.

Reply:
At this stage, the test is written and I'm unsure what it does. As such, I'm inclined to leave the language which I feel doesn't preclude a parallel test, but also doesn't prescribe it.